src/check.rs

```diff
@@ -230,7 +230,7 @@ pub fn check<'a>(pak: &'a Pak, in_file: &mut File, options: CheckOptions) -> Res
                     }
 
                     if let Some(blocks) = record.compression_blocks() {
-                        if !ignore_null_checksums || record.sha1() != &NULL_SHA1 {
+                        if !ignore_null_checksums || &record.sha1().unwrap_or([0u8; 20]) != &NULL_SHA1 {
```

Comment:
Pretty sure these `.unwrap_or([0u8; 20])` calls are inefficient. Looks to me like they're *always* copying 20 bytes to the stack. Hmm. Looks like the optimizer turns it into efficient code anyway? However I think using `.map_or(false, |v| v != NULL_SHA1)` would be nicer? It generates better code in debug mode. See assembly output of this: https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=9cccd68d7e3b2a6f5df5c7d9f3ccc65f

Similar things are below...